vars/dockerExecute.groovy

```diff
@@ -146,60 +150,40 @@ void call(Map parameters = [:], body) {
-                        stashContent: config.stashContent
-                    ]
-
-                    paramMap.containerCommands[config.sidecarImage] = ''
-
-                    paramMap.containerEnvVars[config.dockerImage] = config.dockerEnvVars
-                    paramMap.containerEnvVars[config.sidecarImage] = config.sidecarEnvVars
-
-                    paramMap.containerPullImageFlags[config.dockerImage] = config.dockerPullImage
-                    paramMap.containerPullImageFlags[config.sidecarImage] = config.sidecarPullImage
-
-                    paramMap.containerMap[config.dockerImage] = config.dockerName
-                    paramMap.containerMap[config.sidecarImage] = config.sidecarName
-
-                    paramMap.containerPortMappings = config.containerPortMappings
-
-                    paramMap.containerWorkspaces[config.dockerImage] = config.dockerWorkspace
-                    paramMap.containerWorkspaces[config.sidecarImage] = ''
+                if (!config.dockerName) {
+                    config.dockerName = UUID.randomUUID().toString()
+                }
 
-                    dockerExecuteOnKubernetes(paramMap){
-                        echo "[INFO][${STEP_NAME}] Executing inside a Kubernetes Pod with sidecar container"
-                        if(config.sidecarReadyCommand) {
-                            waitForSidecarReadyOnKubernetes(config.sidecarName, config.sidecarReadyCommand)
-                        }
-                        body()
-                    }
+                Map paramMap = [
+                    script: script,
+                    containerCommands: [:],
+                    containerShell: [:],
+                    containerEnvVars: [:],
+                    containerPullImageFlags: [:],
+                    containerMap: [:],
+                    containerName: config.dockerName,
+                    containerPortMappings: [:],
+                    containerWorkspaces: [:],
+                    stashContent: config.stashContent
+                ]
+
+                paramMap.containerEnvVars[config.dockerImage] = config.dockerEnvVars
+                paramMap.containerPullImageFlags[config.dockerImage] = config.dockerPullImage
+                paramMap.containerMap[config.dockerImage] = config.dockerName
+                paramMap.containerPortMappings = config.containerPortMappings
+                paramMap.containerWorkspaces[config.dockerImage] = config.dockerWorkspace
+                paramMap.containerCommands[config.dockerImage] = config.containerCommand
+                paramMap.containerShell[config.dockerImage] = config.containerShell
+
+                paramMap.sidecarName = parameters.sidecarName
+                paramMap.sidecarImage = parameters.sidecarImage
+                paramMap.sidecarPullImage = parameters.sidecarPullImage
+                paramMap.sidecarReadyCommand = parameters.sidecarReadyCommand
+                paramMap.sidecarEnvVars = parameters.sidecarEnvVars
```

Comment:
This part can be simplified. You should use what was before in the  `if (!config.sidecarImage) {` branch and just add the sidecar line (178 to 182) to the method call, line 150 before.

Reply:
I would loose the containerMap, containerPortMappings and all that stuff?

Reply:
Is done and tests are adapted.